internal/rest/avi_obj_vs.go

```diff
@@ -406,34 +406,53 @@ func (rest *RestOperations) AviVsCacheAdd(rest_op *utils.RestOp, key string) err
 				utils.AviLog.Debug(spew.Sprintf("key: %s, msg: updated VS cache key %v val %v\n", key, k,
 					utils.Stringify(vs_cache_obj)))
 				if svc_mdata_obj.Gateway != "" {
+					updateOptions := status.UpdateOptions{
+						Vip:             vs_cache_obj.Vip,
+						ServiceMetadata: svc_mdata_obj,
+						Key:             key,
+					}
+					statusOption := status.StatusOptions{
+						ObjType: lib.Gateway,
+						Op:      "update",
+						Options: &updateOptions,
+					}
 					if lib.UseServicesAPI() {
-						status.UpdateSvcApiGatewayStatusAddress([]status.UpdateOptions{{
-							Vip:             vs_cache_obj.Vip,
-							ServiceMetadata: svc_mdata_obj,
-							Key:             key,
-						}}, false)
-					} else {
-						status.UpdateGatewayStatusAddress([]status.UpdateOptions{{
-							Vip:             vs_cache_obj.Vip,
-							ServiceMetadata: svc_mdata_obj,
-							Key:             key,
-						}}, false)
+						statusOption.ObjType = lib.SERVICES_API
```

Comment:
@chauhanshubham presume there's no need for service of type LB checks here?

Reply:
lgtm